packages/api/src/hooks/p2p/useUpdateAdvertiser.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useMemo } from 'react';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TPayload = NonNullable<
+    Parameters<ReturnType<typeof useMutation<'p2p_advertiser_update'>>['mutate']>[0]
+>['payload'];
+
+/** A custom hook that updates a P2P advertiser. This can only be used by an approved P2P advertiser.
+ *
+ * To update an advertiser, specify the payload arguments that should be updated, for instance:
+ * @example
+ *  mutate({
+        "is_listed": 0 // optional
+        "upgrade_limits": 1  // optional
+    });
+ *
+*/
+const useUpdateAdvertiser = () => {
+    const invalidate = useInvalidateQuery();
+    const {
+        data,
+        mutate: _mutate,
+        ...rest
+    } = useMutation('p2p_advertiser_update', {
+        onSuccess: () => {
+            invalidate('p2p_advertiser_list');
+        },
+    });
+
+    const mutate = useCallback((payload: TPayload) => _mutate({ payload }), [_mutate]);
+
+    const modified_data = useMemo(() => {
+        const p2p_advertiser_update = data?.p2p_advertiser_update;
+        if (!p2p_advertiser_update) return undefined;
+
+        const { basic_verification, full_verification, is_approved, is_listed, is_online, show_name, created_time } =
+            p2p_advertiser_update;
+
+        return {
+            ...p2p_advertiser_update,
+            basic_verification: Boolean(basic_verification),
+            /** Indicating whether the advertiser's address has been verified. */
+            full_verification: Boolean(full_verification),
+            /** The approval status of the advertiser. */
+            is_approved: Boolean(is_approved),
+            /** Indicates if the advertiser's active adverts are listed. When false, adverts won't be listed regardless if they are active or not. */
+            is_listed: Boolean(is_listed),
+            /** Indicates if the advertiser is currently online. */
+            is_online: Boolean(is_online),
+            /** When true, the advertiser's real name will be displayed on to other users on adverts and orders. */
+            show_name: Boolean(show_name),
```

Comment:
I made modifications for basic_verification and fully_authenticated, but I believe there's no need to change this one. This is because show_name is the variable name.

Reply:
But it dose not show this is a boolean value
maybe you can use should_show_name sth 
what do u think @adrienne-deriv ?

Reply:
we've decided to make show we follow naming conventions for any boolean to have `is`, `has`, etc in the front, so it should follow it as `is_show_name`